### PR TITLE
10bit upgrade

### PR DESCRIPTION
**Pull request by okhlif**
**Description**
10bit upgrade 

- [x]  Intra-inter compound mode
- [x]  Inter-inter compound mode
- [x] Enable Intra PAETH 

Update #622 

**Author**
@okhlif 

**Type of change**
New feature

### CODE REVIEW

**Comment by mafonsoN** [comment]
@okhlif are we expecting gains for 10-bit with these changes?

**Comment by joelsole** [comment]
lgtm - What are the gains?

**Comment by okhlif** [comment]
> @okhlif are we expecting gains for 10-bit with these changes?
> lgtm - What are the gains?

Got 1.69% on 360p objective-fast1 clips ( changed to 10bit) with hbd-md 0.
hbd-md 1  has a run to run difference and a debug mode crash. 
I will enable  Intra-inter compound mode and  Inter-inter compound mode only for hbd-md 0. (i.e. EndDec part) and have a separate project to enable it for hbd-md 1

File: Source/Lib/Encoder/Codec/EbResourceCoordinationProcess.c [hunk 1]
```diff
@@ -690,10 +690,22 @@ void* resource_coordination_kernel(void *input_ptr)
                     sequence_control_set_ptr->static_config.encoder_bit_depth == 8) ? EB_TRUE : EB_FALSE;
 
 #if II_COMP_FLAG
+#if INTER_INTRA_HBD
+            // Set inter-intra mode      Settings
+            // 0                 OFF
+            // 1                 ON
+            sequence_control_set_ptr->seq_header.enable_interintra_compound =  (sequence_control_set_ptr->static_config.encoder_bit_depth == EB_10BIT &&
+                                                                                sequence_control_set_ptr->static_config.enable_hbd_mode_decision ) ? 0:
+                                                                                (sequence_control_set_ptr->static_config.enc_mode == ENC_M0) ? 1 : 0;
```
Comment (mafonsoN) [inline Source/Lib/Encoder/Codec/EbResourceCoordinationProcess.c]: Isn't this line going to turn the interintra_compound tool off for -hbd-md 1 (the default for 10-bit inputs)?

File: Source/Lib/Encoder/Codec/EbResourceCoordinationProcess.c [hunk 2]
```diff
@@ -702,8 +714,14 @@ void* resource_coordination_kernel(void *input_ptr)
             // Set compound mode      Settings
             // 0                 OFF: No compond mode search : AVG only
             // 1                 ON: full
+#if INTER_INTER_HBD
+            sequence_control_set_ptr->compound_mode = (sequence_control_set_ptr->static_config.encoder_bit_depth == EB_10BIT &&
+                                                       sequence_control_set_ptr->static_config.enable_hbd_mode_decision ) ? 0:
+                                                      (sequence_control_set_ptr->static_config.enc_mode <= ENC_M4) ? 1 : 0;
```
Comment (mafonsoN) [inline Source/Lib/Encoder/Codec/EbResourceCoordinationProcess.c]: Same. Isn't this line going to turn the interintra_compound tool off for -hbd-md 1 (the default for 10-bit inputs)?

File: Source/Lib/Encoder/Codec/EbResourceCoordinationProcess.c (hunk 1 above)
Comment (okhlif) [reply Source/Lib/Encoder/Codec/EbResourceCoordinationProcess.c]: Yes we will need to work on it to enable it for -hbd-md 1

**Pull request merged**